core/src/avm2/globals/string.rs

```diff
@@ -126,23 +127,23 @@ fn split<'gc>(
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error> {
     if let Some(this) = this {
-        if matches!(args.get(0).unwrap_or(&Value::Undefined), Value::Undefined) {
-            log::warn!("string.split(undefined) - not implemented");
+        let delimiter = args.get(0).unwrap_or(&Value::Undefined);
+        if matches!(delimiter, Value::Undefined) {
+            let this = Value::from(this);
+            let storage = ArrayStorage::from_args(&[this]);
+            return Ok(ArrayObject::from_storage(activation, storage)
```

Comment:
just to be sure, can't this use something like `iter::once(this).collect()` and then same as below? This way you don't need to go through ArrayStorage manually.

Reply:
Looks like it can, good spot.